File: typescript/difference-of-squares/difference-of-squares.ts
export class Squares {
  private count: number

  constructor(count: number) {
    this.count = count
  }

  get sumOfSquares(): number {
    const n = this.count
    return (n * (n + 1) * (2*n + 1)) / 6
  }

  get squareOfSum(): number {
    const n = this.count
    return ((2 + (n - 1)) * (n / 2)) ** 2
  }

  get difference(): number {
    return this.squareOfSum - this.sumOfSquares
  }
}

File: go/difference-of-squares/difference_of_squares.go
// Package diffsquares provides functions to calculate the difference between the square of the sum and the sum of the squares of the first N natural numbers.
package diffsquares

// SquareOfSum returns the square of the sum of the first N natural numbers.
func SquareOfSum(n int) int {
	sum := (n * (1 + n)) / 2
	return sum * sum
}

// SumOfSquares returns the sum of the squares of the first N natural numbers.
func SumOfSquares(n int) int {
	return int((n * (n + 1) * (2*n + 1)) / 6)
}

// Difference returns the difference between the square of the sum and the sum of the squares of the first N natural numbers.
func Difference(n int) int {
	return SquareOfSum(n) - SumOfSquares(n)
}
